src/Perlang.Parser/NumberParser.cs

```diff
@@ -60,82 +60,121 @@ public static object Parse(NumericToken numericToken)
 
             if (value <= Int32.MaxValue)
             {
-                return (int)value;
+                return new ParsedNumber<int>((int)value);
             }
             else if (value <= UInt32.MaxValue)
             {
-                return (uint)value;
+                return new ParsedNumber<uint>((uint)value);
             }
             else if (value <= Int64.MaxValue)
             {
-                return (long)value;
+                return new ParsedNumber<long>((long)value);
             }
             else if (value <= UInt64.MaxValue)
             {
-                return (ulong)value;
+                return new ParsedNumber<ulong>((ulong)value);
             }
             else // Anything else remains a BigInteger
             {
-                return value;
+                return new ParsedNumber<BigInteger>(value);
```

Comment:
Hmm, but then again... what would be the difference from this `ParsedNumber`, really. :joy: I think I'll just make it simple for now and change it to `LiteralInteger` or something (and then add a separate type for `LiteralFloat`). It should also be a `struct` so that it can use value-type semantics in the CLR.

Reply:
Went with `IntegerLiteral` and `FloatingPointLiteral` as two structs used for this.